src/v/storage/disk_log_impl.cc

```diff
@@ -4116,7 +4116,21 @@ ss::future<> disk_log_impl::remove_kvstore_state(
       .discard_result();
 }
 
-double disk_log_impl::dirty_ratio() const {
+double disk_log_impl::dirty_ratio() {
+    // Sanity check/safety hatch against negative values
+    if (_dirty_segment_bytes < 0 || _closed_segment_bytes < 0) {
+        vlog(
+          stlog.debug,
+          "{}: _dirty_segment_bytes ({}) and/or _closed_segment_bytes ({}) "
+          "have become negative - clearing counters and resumming over {} "
+          "segments in the log.",
+          config().ntp(),
+          _dirty_segment_bytes,
+          _closed_segment_bytes,
+          _segs.size());
+
+        reset_dirty_and_closed_bytes();
```

Comment:
if we log at debug level and then reset, will we ever discover that the stats went negative?